pkg/api/types.go

```diff
@@ -527,8 +527,22 @@ func IsInvalidFilename(name string) bool {
 // When the destination is not specified, the source get copied into current
 // working directory in container.
 func (l *VolumeList) Set(value string) error {
+	volumes := strings.Split(value, ";")
+	newVols := make([]VolumeSpec, len(volumes))
+	for i, v := range volumes {
+		spec, err := l.parseSpec(v)
+		if err != nil {
+			return err
+		}
+		newVols[i] = spec
+	}
+	*l = append(*l, newVols...)
+	return nil
+}
+
+func (l *VolumeList) parseSpec(value string) (VolumeSpec, error) {
 	if len(value) == 0 {
-		return errors.New("invalid format, must be source:destination")
+		return VolumeSpec{}, errors.New("invalid format, must be source:destination")
```

Comment:
should return nil vs. empty

Reply:
Can't do that for value return :/

Reply:
sorry, the implication was to change the return type to a pointer

Reply:
unless there are errors that can be selectively ignored by the caller, why return an empty or partial VolumeSpec ... less space on the stack as well